File: app/src/main/java/com/kksionek/gdzietentramwaj/DataSource/TramData.java
package com.kksionek.gdzietentramwaj.DataSource;

import com.google.android.gms.maps.model.LatLng;
import com.google.gson.annotations.SerializedName;

public class TramData {
    @SerializedName("Time")
    private String mTime;

    @SerializedName("Lat")
    private double mLat;

    @SerializedName("Lon")
    private double mLng;

    @SerializedName("Lines")
    private String mFirstLine;

    @SerializedName("Brigade")
    private String mBrigade;

    private transient LatLng mLatLng = null;

    public String getId() { return mFirstLine + "/" + mBrigade; }

    public String getFirstLine() {
        return mFirstLine;
    }

    public String getTime() {
        return mTime;
    }

    public LatLng getLatLng() {
        if (mLatLng == null) {
            mLatLng = new LatLng(mLat, mLng);
        }
        return mLatLng;
    }
}

File: app/src/main/java/com/kksionek/gdzietentramwaj/dataSource/TramData.kt
package com.kksionek.gdzietentramwaj.dataSource

import com.google.android.gms.maps.model.LatLng
import com.google.gson.annotations.SerializedName

data class TramData(
    @SerializedName("Time")
    val time: String,
    @SerializedName("Lat")
    private val lat: Double,
    @SerializedName("Lon")
    private val lng: Double,
    @SerializedName("Lines")
    val firstLine: String,
    @SerializedName("Brigade")
    private val brigade: String
) {
    val id: String
        get() = "$firstLine/$brigade"

    private var mLatLng: LatLng? = null
    val latLng: LatLng
        get() = mLatLng ?: LatLng(lat, lng).also { mLatLng = it }
}
